torch/csrc/DynamicTypes.cpp

```diff
@@ -63,6 +63,9 @@ PyTypeObject* getPyTypeObject(
   if (storage.device_type() == at::DeviceType::Meta) {
     TORCH_CHECK_NOT_IMPLEMENTED(false, "python bindings for meta storage objects not supported");
   }
+  if (storage.data() == nullptr && storage.nbytes() != 0) {
```

Comment:
we could instead set `set_storage_access_should_throw` https://github.com/pytorch/pytorch/blob/760aefd34d4cb2edb6fb11eea4e72e2d62b51498/c10/core/TensorImpl.h#L2342-L2344 in `THPVariable_make_wrapper_subclass`

Reply:
The thing is, per https://github.com/pytorch/pytorch/issues/65339 in the long term we do want storage represented adequately